### PR TITLE
IDE and Dependencies

### PR DESCRIPTION
**Pull request by jckreg**
- Gradle change to using all external dependencies
- Removing the redundant lib folder
- created a gradle task called "ide-path" that creates the file my.classpath which contains classentry entries for an IDE specific file to use, such as Eclipse, to fix errors, paths, highlighting and auto completion
- added a target to run just the filter aspect of the program

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -42,5 +42,11 @@ kls_database.db
 # IDE specific folders
 # IntelliJ
 .idea/
+# VS Code
 .vscode/
-.dir-local.el
\ No newline at end of file
+.dir-local.el
```
Comment (swflint) [inline .gitignore]:
```suggestion
.dir-locals.el
```

File: .gitignore [hunk 2]
```diff
@@ -42,5 +42,11 @@ kls_database.db
...
+# Eclipse
+.settings/
+.project
+# This one seems to be more widely adopted by many IDEs
+.classpath
```
Comment (swflint) [inline .gitignore:52]: Should this ignore `my.classpath`? That is, the file generated by the ide-paths gradle task?

File: build.gradle.kts [hunk 3]
```diff
@@ -1,64 +1,85 @@
...
-  testImplementation("org.testng:testng:7.10.2")
-  testImplementation("org.junit.jupiter:junit-jupiter-api:5.10.0")
+  testImplementation("org.testng:testng:7.5.1")
+}
+
+tasks.register("ide-paths") {
+  group = "Set-Up"
+  description = "Generate classpath file for IDEs to use for syntax and compiling"
+  doLast {
+    val classpathFile = file("my.classpath")
```
Comment (swflint) [inline build.gradle.kts:43]: Do we want this to generate `my.classpath`? or `.classpath`?

**Review by swflint — CHANGES_REQUESTED** [review]
This seems pretty reasonable, especially assuming all current checks pass.

That said, I see just a couple of things that likely need changed.

File: .gitignore (hunk 2 above)
Comment (jckreg) [reply .gitignore:52]: Probably should ignore that as well. That was only meant to be a temporary file to help people set up an IDE to work better. Good catch

File: build.gradle.kts (hunk 3 above)
Comment (jckreg) [reply build.gradle.kts:43]: I am worried about overwriting .classpath with out any sort of check so I would rather make a my.classpath file and let then rename it or copy what they want. I will change the wording to reflect this

Comment (swflint) [reply build.gradle.kts:43]: Ahh, that makes sense.

**Comment by jckreg** [comment]
Changes requested by @swflint have been added for review.

Does the wording for the ide-path task make sense as to what the goal is?

File: build.gradle.kts [hunk 4]
```diff
@@ -1,64 +1,85 @@
...
+  testImplementation("org.junit.jupiter:junit-jupiter-api:5.11.3")
   testImplementation("org.mockito:mockito-core:5.14.2")
-  testImplementation("org.testng:testng:7.10.2")
-  testImplementation("org.junit.jupiter:junit-jupiter-api:5.10.0")
+  testImplementation("org.testng:testng:7.5.1")
+}
+
+tasks.register("ide-paths") {
+  group = "Set-Up"
+  description = "Generate a file, 'my.classpath'\nFile is a template with class path entries for gradle dependencies"
```
Comment (swflint) [inline build.gradle.kts]:
```suggestion
  description = "Generate a file, 'my.classpath'\nFile is a template with class path entries for gradle-managed dependencies"
```

**Review by swflint — DISMISSED** [review]
I like the changes overall.  I think the suggested wording may be clearer, but will not block over it.

**Comment by jckreg** [comment]
Did a rapid fire commit for the word change using @swflint suggestion

**Review by swflint — APPROVED** [review]

**Pull request merged**